banks-client/src/lib.rs

```diff
@@ -231,6 +246,21 @@ impl BanksClient {
             })
     }
 
+    /// Process a versioned transaction and return the result with metadata.
+    pub fn process_versioned_transaction(
+        &mut self,
+        transaction: VersionedTransaction,
```

Comment:
Rather than taking a `VersionedTransaction` here, how about using something like `SerializableTransaction` from `solana-rpc-client`, and changing all of the implementations use that instead?  Then this could be a function which more specifically returns rich metadata, called `process_transaction_with_metadata`.

I'm not sure of where to refactor `VersionedTransaction` into unfortunately, since sdk doesn't seem like the right place, and it would seem weird to have banks-client depend on rpc-client-api.  It might be fine to just copy the traits into `banks-interface`

Reply:
I updated to `Into<VersionedTransaction>` and `process_transaction_with_metadata`, thanks for the suggestions!

Reply:
Ah, even better, thanks!